internal/docker/container.go

```diff
@@ -342,7 +342,10 @@ func (r *ContainerRunner) collectResults(artifactCfg config.ArtifactDownload, re
 		inProgress--
 
 		jobID := getJobID(res.jobInfo.JobDetailsURL)
-		if download.ShouldDownloadArtifact(jobID, res.passed, res.timedOut, false, artifactCfg) {
+		if download.ShouldDownloadArtifact(jobID, res.passed, res.timedOut, false, artifactCfg.Download) {
+			if artifactCfg.Cleanup {
```

Comment:
How will this behave in case of multi-jobs run ?
From my understanding of the code, it will cleanup for each job (and results in keeping only the last job artifacts of the current run).
Is that the expected behavior ?

Reply:
hmm.. good catch! I think it should delete artifacts before this run and keep all the artifacts for multiple jobs. I'll update this.